opentelemetry-exporters-newrelic/src/test/java/com/newrelic/telemetry/opentelemetry/export/AttributesSupportTest.java

```diff
@@ -0,0 +1,95 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.newrelic.telemetry.Attributes;
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.common.ReadableAttributes;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.resources.Resource;
+import org.junit.jupiter.api.Test;
+
+class AttributesSupportTest {
+
+  @Test
+  void populateLibraryInfo_nullInstrumentationLibrary() {
+    Attributes attributes = new Attributes().put("foo", "bar");
+    Attributes result = AttributesSupport.populateLibraryInfo(attributes, null);
+    assertSame(attributes, result);
+  }
+
+  @Test
+  void populateLibraryInfo_happyPath() {
+    Attributes attributes = new Attributes().put("foo", "bar");
+    Attributes expected =
+        attributes.put(INSTRUMENTATION_NAME, "theName").put(INSTRUMENTATION_VERSION, "theVersion");
+    InstrumentationLibraryInfo libraryInfo = mock(InstrumentationLibraryInfo.class);
+
+    when(libraryInfo.getName()).thenReturn("theName");
+    when(libraryInfo.getVersion()).thenReturn("theVersion");
+
+    Attributes result = AttributesSupport.populateLibraryInfo(attributes, libraryInfo);
+    assertEquals(expected, result);
+  }
+
+  @Test
+  void populateLibraryInfo_nullsAndEmpty() {
+    Attributes attributes = new Attributes().put("foo", "bar");
+    Attributes expected = attributes.copy();
+    InstrumentationLibraryInfo libraryInfo = mock(InstrumentationLibraryInfo.class);
+
+    when(libraryInfo.getName()).thenReturn(null);
+    when(libraryInfo.getVersion()).thenReturn("");
+
+    Attributes result = AttributesSupport.populateLibraryInfo(attributes, libraryInfo);
+    assertEquals(expected, result);
+  }
+
+  @Test
+  void addResourceAttributes_nullResource() {
+    Attributes attributes = new Attributes().put("a", "b");
+    Attributes result = AttributesSupport.addResourceAttributes(attributes, null);
+    assertSame(attributes, result);
+  }
+
+  @Test
+  void addResourceAttributes_happyPath() {
+    Attributes attributes = new Attributes().put("a", "b");
+    ReadableAttributes resourceAttributes =
+        io.opentelemetry.common.Attributes.of(
+            "r1", AttributeValue.stringAttributeValue("v1"),
+            "r2", AttributeValue.longAttributeValue(23),
+            "r3", AttributeValue.booleanAttributeValue(true));
+    Attributes expected =
+        new Attributes().put("a", "b").put("r1", "v1").put("r2", 23).put("r3", true);
+    Resource resource = mock(Resource.class);
+    when(resource.getAttributes()).thenReturn(resourceAttributes);
+    Attributes result = AttributesSupport.addResourceAttributes(attributes, resource);
+    assertSame(attributes, result);
+  }
+
+  @Test
+  void putInAttributes_allTypes() {
+    Attributes attrs = new Attributes().put("y", "z");
+    AttributeValue superBrokenValue = mock(AttributeValue.class);
+    when(superBrokenValue.getType()).thenReturn(AttributeValue.Type.DOUBLE_ARRAY);
```

Comment:
We really should handle array types, but that's a PR for another day.

Also, why not just create an array AttributeValue, rather than mocking it?

Reply:
Why create one, when I can just mock it?  😛 

I'll change it.